mirai-core/src/commonMain/kotlin/contact/GroupSettingsImpl.kt

```diff
@@ -91,9 +92,16 @@ internal class GroupSettingsImpl(
     @Deprecated("Don't use public var internally", level = DeprecationLevel.HIDDEN)
     override var isAnonymousChatEnabled: Boolean
         get() = isAnonymousChatEnabledField
-        @Suppress("UNUSED_PARAMETER")
         set(newValue) {
-            throw UnsupportedOperationException()
+            group.run {
+                checkBotPermission(MemberPermission.ADMINISTRATOR)
+                launch {
+                    //Handle it in NoticePipelineContext#processAllowAnonymousChat
+                    bot.network.run {
+                        SwitchAnonymousChat(bot.client, id, newValue).sendAndExpect()
```

Comment:
更好的处理方式是
```suggestion
                        SwitchAnonymousChat(bot.client, id, newValue).sendAndExpect(bot.network)
```
可以减少一层缩进和 receiver 变换，降低复杂度。不过现在的写法也是可以的。